### ISSUE TITLE
Validation fails due to missing BitsPerComponent

### ISSUE THREAD

**Issue by toreolsensan** [hhrutter/pdfcpu#25]
I'm attempting to merge some files where one file contains the following:
```/Type /XObject
/Subtype /Image
/Width 20
/Height 1
/ImageMask true
/Decode [1 0]
/Length 11 0 R
/Filter /FlateDecode
```

When pdfcpu validates this file, it fails because BitsPerComponent is not specified for this particular XObject.

I'm not an authority on PDF, but in the specification there is a table "Optional parameters for LZWDecode and FlateDecode filters" which has the following description for BitsPerComponent: *(May be used only if Predictor is greater than 1) The number of bits used to represent each colour component in a sample. Valid values are 1, 2, 4, 8, and (PDF 1.5) 16. Default value: 8.*

I think one might argue that since no Predictor is specified BitsPerComponent should actually not be allowed (or at least ignored) for this particular image.

**Comment by hhrutter** [comment]
Are you sure pdfcpu crashes because of a missing `BitsPerComponent` parameter?

Since this XObject does not specify a predictor it should fail with:
 `filter FlateDecode: missing decode parm: Columns`

...which IS a bug!
Thanks for reporting!

Funny - this bug arrives exactly when I was going to work on the remaining unsupported predictors for `FlateDecode`.

**Comment by toreolsensan** [comment]
I successfully processed the PDF in question by changing the relevant line of code in validateImageStreamDictPart2 to `required := OPTIONAL`.

The Predictor key is also in the same table of optional parameters, with a specified default value of 1. Which in turn has the meaning "No prediction" according to "Table 10 - Predictor values" in section 7.4.4.4. I'm reading this from the PDF 32000-1:2008 specification. If you have other sources or sections of the specification that describes how these fields are mandatory I would appreciate a pointer. That would indicate a bug in another library which would be nice to reported.

The file that I'm having problems with contains some data that I don't want exposed to the public, but I'll be happy to share it with you if you are interested. Let me know if and how.

Thanks for your quick response and for sharing this work with the world!

**Comment by hhrutter** [comment]
You are right!
🥇 

So there are 2 locations where we validate the parameters for an Image object.
During validation and during the actual encoding/decoding.
You made the fix at the right location for merging but it is more complicated than that
and it will go in the next release.

There are also changes needed on the filter level which is what I mentioned above.
These all will be fixed together in the next release.

Again thanks for reporting!

I am happy that it worked out for you and please star pdfcpu if you like it and haven't done so already.

**Comment by toreolsensan** [comment]
Glad we got this sorted! I can send you the file for private testing, but I don't want to post a link here. Please can you either post some way to contact you or email me at my-github-username at gmail.com?

**Comment by hhrutter** [comment]
My email is right [here](https://github.com/hhrutter)

**Comment by hhrutter** [comment]
fixed by v0.1.14
